<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>700</width>
    <height>500</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Conway's Game of Life</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="mainLayout">
      <item>
       <layout class="QVBoxLayout" name="gameLayout"/>
      </item>
      <item>
       <layout class="QVBoxLayout" name="setLayout">
        <item>
         <layout class="QHBoxLayout" name="controlLayout">
          <item>
           <widget class="QPushButton" name="startButton">
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="stopButton">
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="clearButton">
            <property name="text">
             <string>Clear</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Universe Size</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="universeSizeLayout">
          <property name="topMargin">
           <number>5</number>
          </property>
          <item>
           <widget class="QSlider" name="cellsControl">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimum">
             <number>50</number>
            </property>
            <property name="maximum">
             <number>1000</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksAbove</enum>
            </property>
            <property name="tickInterval">
             <number>50</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="sizeLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>35</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>50</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="label2">
          <property name="text">
           <string>Generation interval (in msec)</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="iterIntervalLayout">
          <property name="topMargin">
           <number>0</number>
          </property>
          <item>
           <widget class="QSlider" name="iterInterval">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimum">
             <number>10</number>
            </property>
            <property name="maximum">
             <number>500</number>
            </property>
            <property name="sliderPosition">
             <number>100</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="intervalLabel">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>35</width>
              <height>0</height>
             </size>
            </property>
            <property name="text">
             <string>100</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QPushButton" name="colorButton">
          <property name="text">
           <string>Select color for cells</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <widget class="QWidget" name="elements">
           <attribute name="title">
            <string>Elements</string>
           </attribute>
           <widget class="GridTemplateButton" name="pushButton">
            <property name="geometry">
             <rect>
              <x>20</x>
              <y>10</y>
              <width>200</width>
              <height>200</height>
             </rect>
            </property>
           </widget>
          </widget>
          <widget class="QWidget" name="tab">
           <attribute name="title">
            <string>Страница</string>
           </attribute>
          </widget>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>GridTemplateButton</class>
   <extends>QPushButton</extends>
   <header>UI/gridtemplatebutton.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
